Tests/UnitTests/Plugins/Detray/DetrayGeometryConverterTests.cpp

```diff
@@ -0,0 +1,265 @@
+          negZ, CylinderVolumeBounds(50., 140., 100.), RadialBounds(60., 120.),
+          "NegativeEndcap");
+
+  // Declare a barrel sub builder
+  auto barrel0 = std::make_shared<
+      CylindricalVolumeBuilder<CylinderSurface, CylinderBounds>>(
+      Transform3::Identity(), CylinderVolumeBounds(50., 80., 200.),
+      CylinderBounds(65., 180.), "Barrel0");
+
+  // Declare a barrel sub builder
+  auto barrel1 = std::make_shared<
+      CylindricalVolumeBuilder<CylinderSurface, CylinderBounds>>(
+      Transform3::Identity(), CylinderVolumeBounds(80., 110., 200.),
+      CylinderBounds(95., 180.), "Barrel1");
+
+  // Declare a barrel sub builder
+  auto barrel2 = std::make_shared<
+      CylindricalVolumeBuilder<CylinderSurface, CylinderBounds>>(
+      Transform3::Identity(), CylinderVolumeBounds(110., 140., 200.),
+      CylinderBounds(125., 180.), "Barrel2");
+
+  // Create the barrel container builder
+  CylindricalContainerBuilder::Config barrelRCfg;
+  barrelRCfg.builders = {barrel0, barrel1, barrel2};
+  barrelRCfg.binning = {BinningValue::binR};
+
+  auto barrel = std::make_shared<CylindricalContainerBuilder>(
+      barrelRCfg, getDefaultLogger("BarrelBuilderR", Logging::INFO));
+
+  Transform3 posZ = Transform3::Identity();
+  posZ.pretranslate(Vector3(0., 0., 300.));
+  auto endcapP =
+      std::make_shared<CylindricalVolumeBuilder<DiscSurface, RadialBounds>>(
+          posZ, CylinderVolumeBounds(50., 140., 100.), RadialBounds(60., 120.),
+          "PositiveEndcap");
+
+  // Create the barrel container builder
+  CylindricalContainerBuilder::Config barrelEndcapCfg;
+  barrelEndcapCfg.builders = {endcapN, barrel, endcapP};
+  barrelEndcapCfg.binning = {BinningValue::binZ};
+
+  auto barrelEndcap = std::make_shared<CylindricalContainerBuilder>(
+      barrelEndcapCfg, getDefaultLogger("BarrelEndcapBuilder", Logging::INFO));
+
+  // Create the barrel container builder
+  CylindricalContainerBuilder::Config detectorCfg;
+  detectorCfg.builders = {beampipe, barrelEndcap};
+  detectorCfg.binning = {BinningValue::binR};
+
+  auto containerBuilder = std::make_shared<CylindricalContainerBuilder>(
+      detectorCfg, getDefaultLogger("DetectorBuilder", Logging::INFO));
+
+  // Detector builder
+  auto gigConfig = GeometryIdGenerator::Config();
+  auto gig = std::make_shared<GeometryIdGenerator>(gigConfig);
+
+  Acts::Experimental::DetectorBuilder::Config dCfg;
+  dCfg.auxiliary = "*** Test : Cylindrical Detector ***";
+  dCfg.name = "CylindricalDetector";
+  dCfg.builder = containerBuilder;
+  dCfg.geoIdGenerator = gig;
+
+  auto detector = DetectorBuilder(dCfg).construct(tContext);
+
+  // Convert the detector
+  DetrayConversionUtils::GeometryIdCache geoIdCache;
+
+  detray::io::detector_payload payload =
+      DetrayGeometryConverter::convertDetector(geoIdCache, tContext, *detector,
+                                               *logger);
+
+  // test the payload
+  BOOST_CHECK_EQUAL(payload.volumes.size(), 6u);
```

Comment:
Will this get more test cases? That seems to be a lot of setup for only one test case

Reply:
Actually,  yes, particularly the portal splitting test is something I want to do.

Reply:
Added all the portal splitting tests, looks good!